tools/autograd/derivatives.yaml

```diff
@@ -422,6 +426,16 @@
 - name: normal(Tensor self, double mean, double std, Generator generator)
   self: zeros_like(grad)
 
+- name: normal(Tensor mean, double std, Generator generator)
+  mean: grad.type().zeros(mean.sizes())
+
+- name: normal(double mean, Tensor std, Generator generator)
+  std: grad.type().zeros(std.sizes())
+
+- name: normal(Tensor mean, Tensor std, Generator generator)
+  mean: grad.type().zeros(mean.sizes())
+  std: grad.type().zeros(std.sizes())
```

Comment:
I'd say that it's surprising that these functions are differentiable. They either shouldn't be, or should be implemented as native reparametrized functions.

Reply:
This is equivalent to the previous code in `variable.py`

Reply:
(Setting the result to not require grad is equivalent to pushing back zeros)

Reply:
It's not really equivalent. Some operations are illegal on Variables that require grad, and they would work previously

Reply:
Btw, I like the idea re-parameterizing things like normal() (i.e. the VAE trick).  It seems like a better default than zero or raising exceptions

Reply:
The problem is you can't do this for all distributions, so we should probably never backprop through these (and set the results to requires_grad=False, to make it clear that they don't give meaningful gradients). If someone wants reparametrized gradients, use `torch.distributions`